projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.pipes.ts

```diff
@@ -74,7 +74,10 @@ export class IgxPivotRowExpansionPipe implements PipeTransform {
             PivotUtil.flattenHierarchy(data, config, row, expansionStates, defaultExpand, pivotKeys, totalLlv, prevDims, 0, lvl);
             prevDims.push(row);
         }
-        const finalData = config.columnStrategy ? data : data.filter(x => x[pivotKeys.records]);
+        const finalData = config.columnStrategy ? data :
+            enabledRows.length ? data.filter(x => x[pivotKeys.records]) :
+                PivotUtil.flattenHierarchy(data, config, {} as IPivotDimension, expansionStates, defaultExpand, pivotKeys, totalLlv, prevDims, 0, 0);
```

Comment:
Not sure what's the point of flattening the data here.  If there are no row dimensions, there shouldn't be any hierarchy to flatten. There should be just 1 record and it should not have a hierarchy.